### DISCUSSION TITLE
With Next.JS 12.1 the _document is evaluated before _app (ApolloProvider error)

### DISCUSSION THREAD

**Discussion by tomitrescak** [vercel/next.js#36538] [Help]
Hello, is this a feature that on server now _document is evaluated before _app?
I was initialising my <ApolloProvider> in _app but my app broke with upgrade to Next 12.1.
I can see that _document evaluates first with my whole app and _app is left behind as a result I receive invariant error that "client" is not present.

**Comment by icyJoseph** [comment]
Quite possibly related to https://github.com/vercel/next.js/discussions/36533, what do you think? Looks like the same topic.

**Reply by tomitrescak** [reply]
Yes, it does look similar. I've subscribed to it. Let's see if anything will come out of it.
